lib/bundler/stub_specification.rb

```diff
@@ -9,23 +9,86 @@ def self.from_stub(stub)
       spec
     end
 
-    attr_accessor :stub
+    attr_accessor :stub, :ignored
 
     def to_yaml
       _remote_specification.to_yaml
     end
 
+    # @!group Stub Delegates
+
     if Bundler.rubygems.provides?(">= 2.3")
       # This is defined directly to avoid having to load every installed spec
       def missing_extensions?
         stub.missing_extensions?
       end
     end
 
+    def activated
+      stub.activated
+    end
+
+    def activated=(activated)
+      stub.instance_variable_set(:@activated, activated)
+    end
+
+    def default_gem
+      stub.default_gem
+    end
+
+    # This is what we do in bundler/rubygems_ext
```

Comment:
might be worth extracting these two methods into a module we can `include` instead of the copy-paste